hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8748,6 +8863,11 @@ public void startRegionOperation(Operation op) throws IOException {
       throw new NotServingRegionException(getRegionInfo().getRegionNameAsString() + " is closing");
     }
     lock(lock.readLock());
+    // Update regionLockHolders ONLY for any startRegionOperation call that is invoked from an RPC handler
+    Thread thisThread = Thread.currentThread();
+    if (isInterruptableOp) {
+      regionLockHolders.put(thisThread.hashCode(), thisThread);
```

Comment:
Maybe we could just use Collections.newSetFromMap(new ConcurrentHashMap<>()) here?

Reply:
+1

Reply:
Ok